js/stores/windowStore.js

```diff
@@ -750,6 +752,22 @@ const doAction = (action) => {
         }
         break
       }
+    case windowConstants.WINDOW_ON_TOGGLE_BOOKMARK:
+      {
+        // show the AddEditBookmarkHanger control; saving/deleting takes place there
+        const activeFrame = frameStateUtil.getActiveFrame(windowState)
+        let siteDetail = siteUtil.getDetailFromFrame(activeFrame, siteTags.BOOKMARK)
+        const key = siteUtil.getSiteKey(siteDetail)
+
+        if (key !== null) {
```

Comment:
Not sure why this logic should be in `siteUtil.getDetailFromFrame`. In this `if` statement we are getting data from `appStore` and adding them to the `siteDetail`. Not sure where else would we be using this? We use this function in 13 different places. We only want to access `appStore` in reducers and not in components.